tools/run_tests/artifacts/artifact_targets.py

```diff
@@ -118,24 +114,27 @@ def pre_build_jobspecs(self):
     def build_jobspec(self):
         environ = {}
         if self.platform == 'linux_extra':
-            # Raspberry Pi build
-            environ['PYTHON'] = '/usr/local/bin/python{}'.format(
+            # Crosscompilation build for armv7 (e.g. Raspberry Pi)
+            environ['PYTHON'] = '/opt/python/{}/bin/python3'.format(
                 self.py_version)
-            environ['PIP'] = '/usr/local/bin/pip{}'.format(self.py_version)
-            # https://github.com/resin-io-projects/armv7hf-debian-qemu/issues/9
-            # A QEMU bug causes submodule update to freeze, so we copy directly
-            environ['RELATIVE_COPY_PATH'] = '.'
-            # Parallel builds are counterproductive in emulated environment
-            environ['GRPC_PYTHON_BUILD_EXT_COMPILER_JOBS'] = '1'
-            extra_args = ' --entrypoint=/usr/bin/qemu-arm-static '
+            environ['PIP'] = '/opt/python/{}/bin/pip3'.format(self.py_version)
+            environ['GRPC_SKIP_PIP_CYTHON_UPGRADE'] = 'TRUE'
+            environ['GRPC_SKIP_TWINE_CHECK'] = 'TRUE'
+            # when crosscompiling, we need to force statically linking libstdc++
+            # otherwise libstdc++ symbols would be too new and the resulting
+            # wheel wouldn't pass the auditwheel check.
+            # This is needed because C core won't build with GCC 4.8 that's
+            # included in the default dockcross toolchain and we needed
+            # to opt into using a slighly newer version of GCC.
+            environ['GRPC_PYTHON_BUILD_WITH_STATIC_LIBSTDCXX'] = 'TRUE'
+
             return create_docker_jobspec(
                 self.name,
-                'tools/dockerfile/grpc_artifact_linux_{}'.format(self.arch),
+                'tools/dockerfile/grpc_artifact_python_linux_{}'.format(
+                    self.arch),
                 'tools/run_tests/artifacts/build_artifact_python.sh',
                 environ=environ,
-                timeout_seconds=60 * 60 * 7,
-                docker_base_image='quay.io/grpc/raspbian_{}'.format(self.arch),
-                extra_docker_args=extra_args)
+                timeout_seconds=60 * 60)
```

Comment:
I'm wondering if we should have a more lenient timeout to begin with, so we don't risk backsliding on the nightly package distribution. We can watch the build for a few days and tune this timeout lower if we find an hour is sufficient.

Reply:
1hr should really be enough for building  a single python wheel (there's actually running more jobs in parallel, but still). In the past we've been blindly increasing timeout whenever we saw a timeout (instead of investigating and trying to reduce the duration) and it gave us nothing but degradation tower slower and slower CI tests. It didn't actually lead to decreased flakiness since sometimes the build / run has simply just got stuck for interesting reasons and having very high timeouts just prevented us from seeing the actual problem. 
So no, without crystal clear evidence (and a prior genuine attempt to decrease the duration somehow) I'm not really will to increase any timeouts. ;-)

Reply:
Makes sense, I can totally accept those arguments. It's not terribly friendly to the dev community if we set the timeout too restrictively and miss publishing a few nightlies, but it's not the end of the world.